rsvim_core/src/state/ops/message_ops.rs

```diff
@@ -0,0 +1,76 @@
+use crate::state::fsm::StatefulDataAccess;
+use crate::state::ops::cursor_ops::_update_viewport_after_text_changed;
+use crate::ui::tree::Inodeable;
+use crate::ui::viewport::Viewport;
+use crate::ui::widget::command_line::CommandLine;
+use crate::{geo_rect_as, lock};
+
+/// If the message widget has received any text, it prints it in the widget.
+pub fn may_show_message(data_access: &StatefulDataAccess) {
+  {
+    let tree = data_access.tree.clone();
+    let mut tree = lock!(tree);
+    let mut tree_clone = tree.clone();
+    let command_line = tree_clone.command_line_mut().unwrap();
+    let message = command_line.message_mut();
+    if let Some(text) = message.get_message() {
+      let text_contents = message.get_text_contents_mut().upgrade().unwrap();
+      let mut text_contents = lock!(text_contents);
+      let message_content = text_contents.command_line_content_mut();
```

Comment:
Since we also add the "command_line_message" in "TextContents", here we should use it ("command_line_message_mut") instead of "command_line_content_mut".

> "command_line_message" is the text content for messages show in command line, "command_line_content" is the text content for user's typing in command line. We give a text backend data for each of them.